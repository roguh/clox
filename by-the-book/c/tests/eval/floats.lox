// Standard values
print([0.0, 1.0, -1.0, 3.141592653589793, -12345.6789,]);
print([1.2345678901234567e+20, -9.876543210987654e-15,]);

// Edge cases
print([0.0, -0.0,]);
print([Infinity, -Infinity, NaN,]); // Or inf, -inf, nan depending on implementation
print([inf, -inf, nan,]); // Or inf, -inf, nan depending on implementation

print(1.7976931348623157e+308); // Max double value
print(-1.7976931348623157e+308); // Min double value
print(4.9e-324); // Smallest positive subnormal number

// Formatting variations
print([007.12, 123.00, 1.2e+05, 1.2E-03,]);