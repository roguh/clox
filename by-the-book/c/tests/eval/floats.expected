[0, 1, -1, 3.141592653589793, -12345.6789]
[1.234567890123457e+20, -9.876543210987654e-15]
[0, -0]
[inf, -inf, nan]
[inf, -inf, nan]
1.797693134862316e+308
-1.797693134862316e+308
4.940656458412465e-324
[7.12, 123, 120000, 0.0012]
